BigANNBinaryEmbeddingScearioBase.cs

```diff
@@ -132,6 +120,14 @@ protected async Task PerformIngestion(IngestionOperationType ingestionOperationT
             await Task.WhenAll(tasks);
         }
 
+        // check if scenario needs to read label files 
+        protected bool IsWithLabelScenario()
+        {
+            string scenarioName = this.Configurations["AppSettings:scenario:name"] ??
+                throw new ArgumentNullException("AppSettings:scenario:name");
+            return scenarioName.Equals("amazon", StringComparison.OrdinalIgnoreCase);

```

Comment:
we are leaking 'amazon' which is a specific scenario to base class. 
Define an abstact base method and have every scenario derive to True/False.